src/Workspaces/Core/Portable/Rename/ConflictEngine/ConflictResolution.cs

```diff
@@ -57,15 +71,19 @@ internal void ClearDocuments(IEnumerable<DocumentId> conflictLocationDocumentIds
         }
 
         internal void UpdateCurrentSolution(Solution solution)
-            => _newSolution = solution;
+            => NewSolution = solution;
 
-        internal async Task RemoveAllRenameAnnotationsAsync(IEnumerable<DocumentId> documentWithRenameAnnotations, AnnotationTable<RenameAnnotation> annotationSet, CancellationToken cancellationToken)
+        internal async Task<Solution> RemoveAllRenameAnnotationsAsync(
+            Solution intermediateSolution,
```

Comment:
Why do we pass in the intermediate solution rather than just using the NewSolution property?

Reply:
Honestly, because i find that much more confusing.  by stating that it's the NewSolution it indicates to me that it's the 'final solution'.  i.e. the solution after everything is done.  Having it mutate in place is just ocnfusing, so i'm making it s othat the caller keeps track of this temp and passes it along.

Reply:
That's true, but you also renamed it to ModifiedSolution :) It just seems odd that the caller of this method must know what solution should be passed when it's always a property on the type.